### DISCUSSION TITLE
action.json and staged changes

### DISCUSSION THREAD

**Discussion by pabloto** [codefori/vscode-ibmi#528] [Q&A] [answered]
Hi,

When I'm writing a new program, I need to compile only the modified source and not all the source in my repository, when I run the build action if I select staged changes it always compile all of my programs, display file...

How can I compile only the program I'm writing?

Many thanks

**Comment by worksofliam** [comment, marked as answer]
In your actions.json, you can specify the same variables you could use for a steamfile. You could use &BASENAME as a parameter

**Reply by worksofliam** [reply]
Here's my example;

```
[
  {
    "name": "Build and deploy 🔨",
    "command": "/QOpenSys/pkgs/bin/gmake BIN_LIB=&CURLIB OPT=*EVENTF",
    "environment": "pase",
    "deployFirst": true,
    "extensions": [
      "GLOBAL"
    ]
  },
  {
    "name": "Build and deploy current 🔨",
    "command": "/QOpenSys/pkgs/bin/gmake BIN_LIB=&CURLIB OPT=*EVENTF &BASENAME",
    "environment": "pase",
    "deployFirst": true,
    "extensions": [
      "GLOBAL"
    ]
  }
]
```

**Reply by pabloto** [reply]
Hi @worksofliam , tomorrow I'll try and I'll give you a feedback. Many thanks

**Reply by pabloto** [reply]
Hi @worksofliam,

It works perfect.

Many thanks
